neonvm/apis/neonvm/v1/virtualmachine_webhook.go

```diff
@@ -187,6 +195,19 @@ func (r *VirtualMachine) ValidateUpdate(old runtime.Object) error {
 		}
 	}
 
+	// Only allow changing .spec.guest.memoryProvider if it was previously nil, and the new version
+	// is equal to .status.memoryProvider.
+	if !reflect.DeepEqual(before.Spec.Guest.MemoryProvider, r.Spec.Guest.MemoryProvider) {
```

Comment:
Yeah fair enough. Done in 072fe6a130bb1884f79319ec68d0452d92935c4b. Won't squash until the discussion below is resolved, because IIUC we'll have to make this mutable if we put MemoryProvider in the spec, to allow neonvm-controller to set it